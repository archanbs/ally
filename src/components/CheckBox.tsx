import React from "react";

export interface ICheckBoxProps {
  id?: string;
  label: string;
  defaultChecked?: boolean;
  onChange?: () => void;
}

export const CheckBox = ({ label, defaultChecked, id }: ICheckBoxProps) => {
  const [checked, setChecked] = React.useState(defaultChecked || false);

  return (
    <div className="cursor-pointer" onClick={() => setChecked(!checked)}>
      <input className="mr-2 cursor-pointer" type="checkbox" id={id} value={label} />
      <label className="cursor-pointer" htmlFor={id}>
        {label}
      </label>
    </div>
  );
};
