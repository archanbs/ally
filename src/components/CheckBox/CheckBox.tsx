import React from "react";

export interface ICheckBoxProps {
  id?: string;
  label: string;
  defaultChecked?: boolean;
  onChange?: (checked?: boolean) => void;
}

export const CheckBox = ({ label, defaultChecked, id, onChange }: ICheckBoxProps) => {
  const [checked, setChecked] = React.useState(defaultChecked || false);

  return (
    <div
      className="cursor-pointer"
      onClick={() => {
        setChecked(!checked);
        onChange(!checked);
      }}
    >
      <input className="mr-2 cursor-pointer" type="checkbox" id={id} value={label} />
      <label className="cursor-pointer" htmlFor={id}>
        {label}
      </label>
    </div>
  );
};
