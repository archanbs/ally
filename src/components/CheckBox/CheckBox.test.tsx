import renderer from "react-test-renderer";
import { CheckBox } from "./CheckBox";
import React from "react";

test("CheckBox snapshot test", () => {
  const compTree = renderer.create(<CheckBox label="Check box" defaultChecked={true} id="check-box" />).toJSON();
  expect(compTree).toMatchSnapshot();
});

test("CheckBox snapshot test with not checked", () => {
  const compTree = renderer.create(<CheckBox label="Check box" defaultChecked={false} id="check-box" />).toJSON();
  expect(compTree).toMatchSnapshot();
});
