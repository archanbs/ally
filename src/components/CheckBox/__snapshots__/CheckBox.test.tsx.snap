// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CheckBox snapshot test 1`] = `
<div
  className="cursor-pointer"
  onClick={[Function]}
>
  <input
    className="mr-2 cursor-pointer"
    id="check-box"
    type="checkbox"
    value="Check box"
  />
  <label
    className="cursor-pointer"
    htmlFor="check-box"
  >
    Check box
  </label>
</div>
`;

exports[`CheckBox snapshot test with not checked 1`] = `
<div
  className="cursor-pointer"
  onClick={[Function]}
>
  <input
    className="mr-2 cursor-pointer"
    id="check-box"
    type="checkbox"
    value="Check box"
  />
  <label
    className="cursor-pointer"
    htmlFor="check-box"
  >
    Check box
  </label>
</div>
`;
